lib/web/join_tokens.go

```diff
@@ -240,20 +241,35 @@ func createJoinToken(ctx context.Context, m nodeAPIGetter, roles types.SystemRol
 }
 
 func getJoinScript(settings scriptSettings, m nodeAPIGetter) (string, error) {
-	// Skip decoding validation for IAM tokens since they are generated with a different method
-	if settings.joinMethod != string(types.JoinMethodIAM) {
-		// This token does not need to be validated against the backend because it's not used to
-		// reveal any sensitive information. However, we still need to perform a simple input
-		// validation check by verifying that the token was auto-generated.
-		// Auto-generated tokens must be encoded and must have an expected length.
-		decodedToken, err := hex.DecodeString(settings.token)
-		if err != nil {
-			return "", trace.Wrap(err)
+	switch settings.joinMethod {
+	case string(types.JoinMethodUnspecified), string(types.JoinMethodToken), string(types.JoinMethodEC2), string(types.JoinMethodIAM):
```

Comment:
I think `JoinMethodEC2` was never addressed in this code before. I probably need to treat it like `JoinMethodIAM`. @nklaassen can you advise?

Reply:
I don't think we support JoinMethodEC2 at all for this, it only auto-generates IAM tokens, can probably just reject it